Eos/Soave-Redlich-Kwong/EOS.H

```diff
@@ -363,129 +399,76 @@ EY2T(amrex::Real E, amrex::Real Y[], amrex::Real& T)
   amrex::Error("EY2T not physically possible for this EoS");
 }
 
+
 AMREX_GPU_HOST_DEVICE
 AMREX_FORCE_INLINE
 void
 REY2T(amrex::Real R, amrex::Real E, amrex::Real Y[], amrex::Real& T)
 {
-  amrex::Real tau = 1.0/R;
   amrex::Real Tn = T;
   amrex::Real bm, am, dAmdT, d2AmdT2;
   amrex::Real K1, ei_ig[NUM_SPECIES], Eig, Cv, dT;
   int nIter = 0;
-  int lierr = 0;
   amrex::Real fzero = 1.0;
   
-  // Run some tests
-  //  First print inputs
-  Tn = 501.0;
-  //std::cout << "Testing" <<std::endl;
-  //std::cout << "R " << R << std::endl;
-  //std::cout << "E " << E << std::endl;
-  //std::cout << "Y ";
-  for (int ii =0; ii<NUM_SPECIES; ii++) { std::cout << Y[ii]; };
-  //std::cout << std::endl;
-  //std::cout << "T " << Tn << std::endl;
-
-  // Now run tests
-  //std::cout << " Running tests:" << std::endl;
-  // EOS::MixingRuleBm(Y, bm);  
-  // std::cout << "bm " << bm << std::endl;
-  // EOS::MixingRuleAm(Tn, Y, am);
-  // std::cout << "am " << am << std::endl;
-  // EOS::MixingRuleAmBm(Tn, Y, am, bm);
-  // std::cout << "ambm " << am << " " << bm << std::endl;
-  // EOS::Calc_dAmdT(Tn, Y, dAmdT);
-  // std::cout << "dAmdT " << dAmdT << std::endl;
-  // EOS::Calc_d2AmdT2(Tn, Y, d2AmdT2);
-  // std::cout << "d2AmdT2" << d2AmdT2 << std::endl;
-  
-  // amrex::Real temp_vec[NUM_SPECIES];
-  // EOS::Calc_dAmdY(Tn, Y, temp_vec);
-  // std::cout << "dAmdY ";
-  // for (int ii =0; ii<NUM_SPECIES; ii++) { std::cout << temp_vec[ii] << " "; };
-  // std::cout << std::endl;
-  
-  // EOS::Calc_d2AmdTY(Tn, Y, temp_vec);
-  // std::cout << "d2AmdTY ";
-  // for (int ii =0; ii<NUM_SPECIES; ii++) { std::cout << temp_vec[ii] << " "; };
-  // std::cout << std::endl;
-  
-  // amrex::Real temp_vec2[NUM_SPECIES][NUM_SPECIES];
-  // EOS::Calc_d2AmdY2(Tn, Y, temp_vec2);
-  // std::cout << "d2AmdTY ";
-  // for (int ii =0; ii<NUM_SPECIES; ii++)
-  //   {for (int jj =0; jj<NUM_SPECIES; jj++){
-  // 	std::cout << temp_vec2[ii][jj] << " ";
-  //     }}
-  // std::cout << std::endl;
-  // std::cout << std::endl;
-  
   EOS::MixingRuleBm(Y, bm);
-  K1 = (1.0 / bm) * log(1.0 + bm/tau); // log or log10?	\
+  K1 = (1.0 / bm) * log(1.0 + bm*R); 
 
-  // Use input T as initial guess, even though it may be garbage
+  // Use input T as initial guess, but constrain it to reasonable bounds
+  Tn = amrex::min(amrex::max(T,300.0),5000.0);
   
   while ( amrex::Math::abs(fzero) > convCrit && nIter < maxIter) {
     nIter++;
-    EOS::MixingRuleAm(Tn, Y, am);
-    EOS::Calc_dAmdT(Tn, Y, dAmdT);
-    EOS::Calc_d2AmdT2(Tn, Y, d2AmdT2);
+    EOS::Calc_Am_and_derivs(Tn, Y, am, dAmdT, d2AmdT2);
+    
     // ideal gas internal energy
     CKUMS(&Tn, ei_ig);
     Eig = 0.0; for (int ii = 0; ii< NUM_SPECIES; ii++) {Eig += Y[ii]*ei_ig[ii];};
     // ideal gas heat capacity
     CKCVBS(&Tn, Y, &Cv);
-    //std::cout << "Cv1 " <<Cv << std::endl;
-    //std::cout << "Eig " << Eig << std::endl;
     // real gas heat capacity
-    Cv -= Tn * d2AmdT2 * K1;
+    Cv += Tn * d2AmdT2 * K1;
```

Comment:
I assume this sign change is correct?  I have no idea, but it looked scary enough to call out.

Reply:
The sign change matches other places in the code where Cv is calculated. Here Cv is just used as the derivative in Newton iteration so this bug just slowed down convergence but didn't affect the final result. With the change, iterations to convergence go from ~100 to 5 for some cases.

Reply:
If the number of iterations dropped by that much, my guess is that the sign is consistent now with the formulation of the residual that is being driven to zero by the solve, and Jacobian that is being evaluated for it.  5 iterations might well be showing the expected quadratic convergence, whereas 100 can't possibly be in the asymptotic region and so suggests a bug (or a very unfortunate discontinuity at the temperature switch points in the pure Cv fits!).  It would be satisfying to comment out the temperature switches and verify quadratic convergence.